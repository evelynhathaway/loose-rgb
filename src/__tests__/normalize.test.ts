import {normalize} from "../index.js";


describe("normalize", () => {
	const normalizeEdgeCases = [
		{r: 1, g: 2, b: 3},
		{r: 1, g: 2, b: 3, a: 0.4},
		{R: 1, G: 2, B: 3},
		{R: 1, G: 2, B: 3, A: 0.4},
		{red: 1, green: 2, blue: 3},
		{red: 1, green: 2, blue: 3, alpha: 0.4},
		[1, 2, 3],
		[1, 2, 3, 0.4],
		"rgb(0, 213, 34)",
		"rgba(0, 213, 34,.12)",
		"#fff",
		"#000000",
		"#FFBB00AA",
		"123",
		"FFFF",
	];

	test.each(normalizeEdgeCases)("%p", (color) => {
		expect(normalize(color)).toMatchSnapshot();
	});

	test("throw error if not recognized", () => {
		expect(() => {
			(normalize as CallableFunction)({r: 1, a: "string"});
		}).toThrowError("The input into `looseRGB.normalize` wasn't a recognized RGB type.");
	});
});
