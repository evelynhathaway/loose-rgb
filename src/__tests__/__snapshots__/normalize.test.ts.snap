// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize "#000000" 1`] = `
Object {
  "b": 0,
  "g": 0,
  "r": 0,
}
`;

exports[`normalize "#FFBB00AA" 1`] = `
Object {
  "a": 0.66796875,
  "b": 0,
  "g": 187,
  "r": 255,
}
`;

exports[`normalize "#fff" 1`] = `
Object {
  "b": 255,
  "g": 255,
  "r": 255,
}
`;

exports[`normalize "123" 1`] = `
Object {
  "b": 51,
  "g": 34,
  "r": 17,
}
`;

exports[`normalize "FFFF" 1`] = `
Object {
  "a": 1,
  "b": 255,
  "g": 255,
  "r": 255,
}
`;

exports[`normalize "rgb(0, 213, 34)" 1`] = `
Object {
  "b": 34,
  "g": 213,
  "r": 0,
}
`;

exports[`normalize "rgba(0, 213, 34,.12)" 1`] = `
Object {
  "a": 0.12,
  "b": 34,
  "g": 213,
  "r": 0,
}
`;

exports[`normalize [1, 2, 3, 0.4] 1`] = `
Object {
  "a": 0.4,
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize [1, 2, 3] 1`] = `
Object {
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize {"A": 0.4, "B": 3, "G": 2, "R": 1} 1`] = `
Object {
  "a": 0.4,
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize {"B": 3, "G": 2, "R": 1} 1`] = `
Object {
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize {"a": 0.4, "b": 3, "g": 2, "r": 1} 1`] = `
Object {
  "a": 0.4,
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize {"alpha": 0.4, "blue": 3, "green": 2, "red": 1} 1`] = `
Object {
  "a": 0.4,
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize {"b": 3, "g": 2, "r": 1} 1`] = `
Object {
  "b": 3,
  "g": 2,
  "r": 1,
}
`;

exports[`normalize {"blue": 3, "green": 2, "red": 1} 1`] = `
Object {
  "b": 3,
  "g": 2,
  "r": 1,
}
`;
